service/controller/v1/resource/chart/delete.go

```diff
@@ -3,33 +3,28 @@ package chart
 import (
 	"context"
 	"fmt"
-	"reflect"
 
-	"github.com/giantswarm/chart-operator/service/controller/v1/key"
+	"k8s.io/helm/pkg/helm"
+
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
-	customObject, err := key.ToCustomObject(obj)
-	if err != nil {
-		return microerror.Mask(err)
-	}
```

Comment:
Boyscouting: We actually don't need the custom object here, since we can delete by ReleaseName of the `deleteChange`.